pkg/frontend/querymiddleware/astmapper/instant_splitting_stats.go

```diff
@@ -2,8 +2,19 @@
 
 package astmapper
 
+type NoOpReason string
+
+// Possible noop reasons
+const (
+	SmallIntervalNoOpReason = NoOpReason("small_interval")
+	SubqueryNoOpReason      = NoOpReason("subquery")
+	NonSplittableNoOpReason = NoOpReason("non_splittable")
+	noneNoOpReason          = NoOpReason("none")
+)
+
 type InstantSplitterStats struct {
-	splitQueries int
+	splitQueries    int        // counter of split queries (0 if non-splittable)
+	noOpQueryReason NoOpReason // reason the initial query is a no operation
```

Comment:
[nit] In the metrics we call it it "skipped" instead of "noop". Should we call it "skipped" here for consistency too? Also why not just using the same constant values between the two?

Reply:
Yes, makes sense. Addressed in: https://github.com/grafana/mimir/pull/2642/commits/281a20ad817f41ecd5202797cd31cc296b9565da